### ISSUE TITLE
[Feature Request] Tag aliasing

### ISSUE THREAD

**Issue by JetstreamLarry** [NO-ob/LoliSnatcher_Droid#137]
There are a few reoccurring scenarios in which having the ability to alias tags (like you would with aliasing commands for a shell) would be useful:

- **Different boorus using different tags for the same thing**: for example, when referencing characters from Touhou, danbooru uses the tag `touhou`, while sankaku uses `touhou_project`. Aliasing `touhou` and `touhou_project` to `touhou*` or `~touhou ~touhou_project` would make quickly switching between boorus (or multibooru) less of a pain. Another example is name order, such as`reimu_hakurei` vs. `hakurei_reimu`.
- **Multiple tags referencing the same thing**: some less organized boorus may use `animated`, `video`, `gif` and/or other similar tags on animated content without much regularity. Aliasing `animated` to `~animated ~video ~gif` could solve this.
- **Niche tags**: users may want to look for images with some specific characteristics that are not explicitly tagged, but are easily expressed by combining two or more tags. For example, aliasing `zun_art` to `touhou official_style` (which is already present in some boorus, but not consistently tagged) (also, alias inside of an alias).
- **Grouping up related tags**: for example, if a user wants to search images of the main mobile suits from ZZ Gundam, they could alias  `gundam_team` to `zz_gundam zeta_gundam gundam_mk_ii hyaku_shiki`.

Aliases could also be made to be global or to only apply to specific boorus, since tagging is often inconsistent between sites.

**Comment by NANI-SORE** [comment]
Good idea, I have problem with that myself 😅, when I have to change from ```sound``` to ```has_audio``` on sankaku.
Maybe we'll get around to it somewhere down the line after we release tag types (soon ™) and rework some related systems (i.e. tag suggestions).
But filling those aliases will be completely on the user, we won't distribute any curated lists, like we plan to do with tag types.

**Comment by coolst3r** [comment]
To make an alias global, it would need to be recognized by all platforms or applications that use the same tagging system. This can be achieved by creating the alias within a centralized database or repository that is accessible to all platforms that use the tagging system. For example, if multiple boorus are using the same tagging system, a centralized database or repository can be used to store and manage the aliases, which can then be accessed and utilized by all of the boorus. Alternatively, if the tagging system is used across multiple applications, a similar approach can be used to create and manage global aliases that can be used across all of the applications.
